questionnaire/questionnaire-service.js

```diff
@@ -588,18 +586,131 @@ function createQuestionnaireService({
         };
     }
 
+    async function createSubmission(questionnaireId) {
+        // 1) get questionnaire instance.
+        const questionnaire = await getQuestionnaire(questionnaireId);
+        if (!questionnaire) {
+            throw new VError(
+                {
+                    name: 'ResourceNotFound'
+                },
+                `Questionnaire with questionnaireId "${questionnaireId}" does not exist`
+            );
+        }
+
+        // 2) get questionnaire instance's submission status.
+        const submissionStatus = await getQuestionnaireSubmissionStatus(questionnaireId);
+        // 3) are we currently, or have we been on this questionnaire's summary page?
+        // we infer a questionnaire is complete if the user has visited the summary page.
+        const isQuestionnaireComplete = questionnaire.progress.includes(
+            questionnaire.routes.summary
+        );
+
+        if (!isQuestionnaireComplete) {
+            throw new VError(
+                {
+                    name: 'ResourceConflict'
+                },
+                `Questionnaire with ID "${questionnaireId}" is not in a submittable state`
+            );
+        }
+
+        // if the submission status is anything other than 'NOT_STARTED' then it
+        // means that the submission resource has been previously created.
+        // also skip over this for failed application so they can be resubmitted.
+        if (!['NOT_STARTED', 'FAILED'].includes(submissionStatus)) {
+            throw new VError(
+                {
+                    name: 'ResourceConflict'
+                },
+                `Submission resource with ID "${questionnaireId}" already exists`
+            );
+        }
```

Comment:
Is this logic definitely correct/necessary? We don't want it to try to submit anything that's not ready to submit, but these cases shouldn't be returned by the query anyway. Also, should we be limiting the type of status we can resubmit? I'm sure we discussed this, but while it **_shouldn't_** be able to get stuck `IN_PROGRESS` anymore, but what if it does? Was it agreed we ignore this?